### DISCUSSION TITLE
Rate Limiter

### DISCUSSION THREAD

**Discussion by platinumachievements** [isFakeAccount/psnawp#75] [Ideas]
ive found thta im making a massive amout of calls to retrieve all the trophy details. im trying to implement a rate limiter with sessions. could a simple method be incorporated into the api, maybe using requests_ratelimiter, i think i got 'banned' for a while as i did over 300 ish calls every 15 minutes.

**Comment by isFakeAccount** [comment]
Hmm. Good point. Also I should add some warning in readme so people don't make bulk api requests.

**Reply by platinumachievements** [reply]
Yeah, it wasnt until I ran some debug code and actually saw how many calls the api has to make to get all the information required. I know it would need a fair amount but when you have over 100 games  and you need the trophies it hits the limit pretty quick!

**Comment by isFakeAccount** [comment]
v2.1.0 now uses rate limiter. It rate limits at 300 requests per 15 minutes.
